libgadget/gravpm.c

```diff
@@ -92,9 +93,14 @@ gravpm_force(PetaPM * pm, DomainDecomp * ddecomp, Cosmology * CP, double Time, d
     }
 
     /* Tree freed in PM*/
-    ForceTree Tree = {0};
+    // ForceTree Tree = {0};
+    ForceTree * Tree_ptr;
```

Comment:
Let's make this stack-allocated again, but pass it by value (and probably const).